scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -258,99 +255,81 @@ private int selectAndStartTasks(Policy currentPolicy, Map<JobId, JobDescriptor>
             updateVariablesForTasksToSchedule(taskRetrievedFromPolicy);
 
             for (EligibleTaskDescriptor etd : taskRetrievedFromPolicy) {
+                final InternalTask internalTask = ((EligibleTaskDescriptorImpl) etd).getInternal();
                 // load and Initialize the executable container
-                loadAndInit(((EligibleTaskDescriptorImpl) etd).getInternal());
+                loadAndInit(internalTask);
             }
 
-            while (!taskRetrievedFromPolicy.isEmpty()) {
-
-                if (freeResources.isEmpty()) {
-                    break;
-                }
+            while (!taskRetrievedFromPolicy.isEmpty() && !freeResources.isEmpty()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)